Assignment 3/src/main/java/com/nuclei/assignment/service/menuoptions/MenuOptionsImpl.java

```diff
@@ -0,0 +1,221 @@
+          System.out.println(exception.getMessage());
+        } finally {
+          System.out.println(StringConstantsUtils.DIVIDER);
+          System.out.print(StringConstantsUtils.CONTINUE_MORE_OPERATIONS);
+          addMoreUsers = scanner.nextLine().toLowerCase(Locale.ROOT);
+        }
+      }
+    } catch (Exception exception) {
+      System.out.println(exception.getMessage());
+    }
+  }
+  
+  // Suppressing warning of
+  // The method 'chooseMenuOption(Scanner)' has a cyclomatic complexity of 10.
+  // Since the menu has 8 options, so it can't be reduced further.
+  @SuppressWarnings("PMD.CyclomaticComplexity")
+  private void chooseMenuOption(final Scanner scanner) throws CustomException {
+    final int menuOption = Integer.parseInt(scanner.nextLine());
+    switch (menuOption) {
+      case 1:
+        getImmediateParents(scanner);
+        break;
+      case 2:
+        getImmediateChildren(scanner);
+        break;
+      case 3:
+        getAllAncestors(scanner);
+        break;
+      case 4:
+        getAllDescendants(scanner);
+        break;
+      case 5:
+        deleteDependency(scanner);
+        break;
+      case 6:
+        deleteUserById(scanner);
+        break;
+      case 7:
+        createDependency(scanner);
+        break;
+      case 8:
+        addUserOption(scanner);
+        break;
+      default:
+        logger.error(String.format(ExceptionsConstantsUtils.INVALID_INPUT, menuOption));
+        throw new CustomException(String.format(ExceptionsConstantsUtils.INVALID_INPUT,
+            menuOption));
+    }
+  }
+  
+  private void addUserOption(final Scanner scanner) throws CustomException {
+    createUser(scanner);
+    if (graphOperations.getAllUsers().size()
+        >= ExceptionsConstantsUtils.MUST_HAVE_USERS_BEFORE_DEPENDENCY_FORMATION) {
+      createDependency(scanner);
+    }
+  }
+  
+  private void createUser(final Scanner scanner) throws CustomException {
+    System.out.println(StringConstantsUtils.ADD_USER_ID);
+    final int id = parser.parseId(scanner.nextLine());
+    if (graphOperations.checkUserExistById(id)) {
+      logger.error(String.format(ExceptionsConstantsUtils.ID_ALREADY_EXISTS,
+          id));
+      throw new CustomException(String.format(ExceptionsConstantsUtils.ID_ALREADY_EXISTS,
+          id));
+    }
+    
+    System.out.println(StringConstantsUtils.ADD_USER_NAME);
+    final String name = parser.parseName(scanner.nextLine());
+    final Map<String, String> userDetails = new ConcurrentHashMap<>();
+    
+    System.out.println(StringConstantsUtils.CHECK_USER_DETAILS);
+    String addUserDetails = parser.parseName(scanner.nextLine());
+    while (addUserDetails.equals(StringConstantsUtils.CONFIRMATION)) {
+      System.out.println(StringConstantsUtils.ADD_USER_DETAILS);
+      final String[] details = parser.parseDetails(scanner.nextLine());
+      userDetails.put(details[0], details[1]);
+      
+      System.out.println(StringConstantsUtils.CHECK_USER_DETAILS);
```

Comment:
rename it to something which means confirmation?

Reply:
done